--- a/crt/s2n/CMakeLists.txt	1985-10-26 17:15:00.000000000 +0900
+++ b/crt/s2n/CMakeLists.txt	2020-12-05 12:42:21.000000000 +0900
@@ -193,7 +193,7 @@
 
 set(CMAKE_C_FLAGS_DEBUGOPT "")
 
-target_compile_options(${PROJECT_NAME} PRIVATE -pedantic -std=gnu99 -Wall -Werror -Wimplicit -Wunused -Wcomment -Wchar-subscripts
+target_compile_options(${PROJECT_NAME} PRIVATE -pedantic -std=gnu99 -Wall -Wimplicit -Wunused -Wcomment -Wchar-subscripts
         -Wuninitialized -Wshadow -Wcast-qual -Wcast-align -Wwrite-strings -Wno-deprecated-declarations -Wno-unknown-pragmas -Wformat-security
         -Wno-missing-braces -fvisibility=hidden -DS2N_EXPORTS)
 
diff -urN a/crt/s2n/s2n.mk b/crt/s2n/s2n.mk
--- a/crt/s2n/s2n.mk	1985-10-26 17:15:00.000000000 +0900
+++ b/crt/s2n/s2n.mk	2020-12-05 12:42:49.000000000 +0900
@@ -42,7 +42,7 @@
 	DEFAULT_CFLAGS = -std=c99 -Wcast-qual
 endif
 
-DEFAULT_CFLAGS += -pedantic -Wall -Werror -Wimplicit -Wunused -Wcomment -Wchar-subscripts -Wuninitialized \
+DEFAULT_CFLAGS += -pedantic -Wall -Wimplicit -Wunused -Wcomment -Wchar-subscripts -Wuninitialized \
                  -Wshadow  -Wcast-align -Wwrite-strings -fPIC -Wno-missing-braces\
                  -D_POSIX_C_SOURCE=200809L -O2 -I$(LIBCRYPTO_ROOT)/include/ \
                  -I$(S2N_ROOT)/api/ -I$(S2N_ROOT) -Wno-deprecated-declarations -Wno-unknown-pragmas -Wformat-security \
