diff -urN a/deps/v8/src/compiler/backend/mips/code-generator-mips.cc b/deps/v8/src/compiler/backend/mips/code-generator-mips.cc
--- a/deps/v8/src/compiler/backend/mips/code-generator-mips.cc	2019-08-07 05:46:24.000000000 +0900
+++ b/deps/v8/src/compiler/backend/mips/code-generator-mips.cc	2019-08-07 15:21:09.743401341 +0900
@@ -171,7 +171,9 @@
       __ Push(ra);
     }
 
-    if (stub_mode_ == StubCallMode::kCallWasmRuntimeStub) {
+    if (mode_ == RecordWriteMode::kValueIsEphemeronKey) {
+      __ CallEphemeronKeyBarrier(object_, scratch1_, save_fp_mode);
+    } else if (stub_mode_ == StubCallMode::kCallWasmRuntimeStub) {
       // A direct call to a wasm runtime stub defined in this module.
       // Just encode the stub index. This will be patched when the code
       // is added to the native module and copied into wasm code space.
diff -urN a/deps/v8/src/mips/assembler-mips.cc b/deps/v8/src/mips/assembler-mips.cc
--- a/deps/v8/src/mips/assembler-mips.cc	2019-08-07 05:46:24.000000000 +0900
+++ b/deps/v8/src/mips/assembler-mips.cc	2019-08-07 15:06:35.434169245 +0900
@@ -3858,7 +3858,7 @@
     if (imm == kEndOfJumpChain) {
       return;
     }
-    imm += pc_delta;
+    imm -= pc_delta;
     DCHECK_EQ(imm & 3, 0);
     PatchLuiOriImmediate(pc, imm, instr1, 0 * kInstrSize, instr2, ori_offset);
     return;
diff -urN a/deps/v8/src/mips/interface-descriptors-mips.cc b/deps/v8/src/mips/interface-descriptors-mips.cc
--- a/deps/v8/src/mips/interface-descriptors-mips.cc	2019-08-07 05:46:24.000000000 +0900
+++ b/deps/v8/src/mips/interface-descriptors-mips.cc	2019-08-07 15:06:35.434169245 +0900
@@ -61,6 +61,18 @@
   data->InitializePlatformSpecific(kParameterCount, default_stub_registers);
 }
 
+void EphemeronKeyBarrierDescriptor::InitializePlatformSpecific(
+    CallInterfaceDescriptorData* data) {
+  const Register default_stub_registers[] = {a0, a1, a2, a3, kReturnRegister0};
+
+  data->RestrictAllocatableRegisters(default_stub_registers,
+                                     arraysize(default_stub_registers));
+
+  CHECK_LE(static_cast<size_t>(kParameterCount),
+           arraysize(default_stub_registers));
+  data->InitializePlatformSpecific(kParameterCount, default_stub_registers);
+}
+
 const Register FastNewFunctionContextDescriptor::ScopeInfoRegister() {
   return a1;
 }
diff -urN a/deps/v8/src/mips/macro-assembler-mips.cc b/deps/v8/src/mips/macro-assembler-mips.cc
--- a/deps/v8/src/mips/macro-assembler-mips.cc	2019-08-07 05:46:24.000000000 +0900
+++ b/deps/v8/src/mips/macro-assembler-mips.cc	2019-08-07 15:06:35.434169245 +0900
@@ -251,6 +251,32 @@
   MultiPop(regs);
 }
 
+void TurboAssembler::CallEphemeronKeyBarrier(Register object, Register address,
+                                             SaveFPRegsMode fp_mode) {
+  EphemeronKeyBarrierDescriptor descriptor;
+  RegList registers = descriptor.allocatable_registers();
+
+  SaveRegisters(registers);
+
+  Register object_parameter(
+      descriptor.GetRegisterParameter(EphemeronKeyBarrierDescriptor::kObject));
+  Register slot_parameter(descriptor.GetRegisterParameter(
+      EphemeronKeyBarrierDescriptor::kSlotAddress));
+  Register fp_mode_parameter(
+      descriptor.GetRegisterParameter(EphemeronKeyBarrierDescriptor::kFPMode));
+
+  Push(object);
+  Push(address);
+
+  Pop(slot_parameter);
+  Pop(object_parameter);
+
+  Move(fp_mode_parameter, Smi::FromEnum(fp_mode));
+  Call(isolate()->builtins()->builtin_handle(Builtins::kEphemeronKeyBarrier),
+       RelocInfo::CODE_TARGET);
+  RestoreRegisters(registers);
+}
+
 void TurboAssembler::CallRecordWriteStub(
     Register object, Register address,
     RememberedSetAction remembered_set_action, SaveFPRegsMode fp_mode) {
diff -urN a/deps/v8/src/mips/macro-assembler-mips.h b/deps/v8/src/mips/macro-assembler-mips.h
--- a/deps/v8/src/mips/macro-assembler-mips.h	2019-08-07 05:46:24.000000000 +0900
+++ b/deps/v8/src/mips/macro-assembler-mips.h	2019-08-07 15:06:35.438169232 +0900
@@ -329,6 +329,8 @@
   void CallRecordWriteStub(Register object, Register address,
                            RememberedSetAction remembered_set_action,
                            SaveFPRegsMode fp_mode, Address wasm_target);
+  void CallEphemeronKeyBarrier(Register object, Register address,
+                               SaveFPRegsMode fp_mode);
 
   // Push multiple registers on the stack.
   // Registers are saved in numerical order, with higher numbered registers
