diff -urN a/configure.py b/configure.py
--- a/configure.py	2019-08-27 00:00:31.000000000 +0900
+++ b/configure.py	2019-08-30 14:15:19.484866116 +0900
@@ -930,6 +930,7 @@
   can_use_fpu_instructions = (options.mips_float_abi != 'soft')
   o['variables']['v8_can_use_fpu_instructions'] = b(can_use_fpu_instructions)
   o['variables']['v8_use_mips_abi_hardfloat'] = b(can_use_fpu_instructions)
+  o['variables']['v8_enable_embedded_builtins'] = 1 if options.without_snapshot else 0
   o['variables']['mips_arch_variant'] = options.mips_arch_variant
   o['variables']['mips_fpu_mode'] = options.mips_fpu_mode
 
