--- a/tools/v8_gypfiles/v8.gyp	2019-06-27 19:12:20.000000000 +0900
+++ b/tools/v8_gypfiles/v8.gyp	2019-07-01 14:40:48.292020880 +0900
@@ -1323,6 +1323,7 @@
     {
       'target_name': 'bytecode_builtins_list_generator',
       'type': 'executable',
+      'libraries!':[ '-licui18n', '-licuuc', '-licudata', '-lcrypto', '-lssl', '-lz' ],
       'conditions': [
         ['want_separate_host_toolset', {
           'toolsets': ['host'],
@@ -1346,6 +1347,8 @@
     {
       'target_name': 'mksnapshot',
       'type': 'executable',
+      'libraries!':[ '-lcrypto', '-lssl', '-lz' ],
+      'libraries':[ '-L../../../../staging_dir/hostpkg/share/icu/64.2/lib' ],
       'dependencies': [
         'v8_base_without_compiler',
         'v8_compiler_for_mksnapshot',
@@ -1368,6 +1371,7 @@
     {
       'target_name': 'torque',
       'type': 'executable',
+      'libraries!':[ '-licui18n', '-licuuc', '-licudata', '-lcrypto', '-lssl', '-lz' ],
       'dependencies': [
         'torque_base',
         # "build/win:default_exe_manifest",
@@ -1406,6 +1410,7 @@
     {
       'target_name': 'torque-language-server',
       'type': 'executable',
+      'libraries!':[ '-licui18n', '-licuuc', '-licudata', '-lcrypto', '-lssl', '-lz' ],
       'conditions': [
         ['want_separate_host_toolset', {
           'toolsets': ['host'],
@@ -1433,6 +1438,9 @@
     {
       'target_name': 'gen-regexp-special-case',
       'type': 'executable',
+      'libraries!':[ '-lcrypto', '-lssl', '-lz' ],
+      'libraries':[ '-Wl,-rpath ../../../../staging_dir/hostpkg/share/icu/64.2/lib' ],
+      'libraries':[ '-Wl,-rpath-link ../../../../staging_dir/hostpkg/share/icu/64.2/lib' ],
       'dependencies': [
         'v8_libbase',
         # "build/win:default_exe_manifest",
