--- a/src/ffi.cc
+++ b/src/ffi.cc
@@ -92,10 +92,11 @@
   SET_ENUM_VALUE(FFI_WIN64);
 #elif defined __aarch64__
   SET_ENUM_VALUE(FFI_SYSV);
-#else
-  /* ---- Intel x86 and AMD x86-64 - */
+#elif defined(X86);
+  /* ---- Intel x86 ---- */
   SET_ENUM_VALUE(FFI_SYSV);
-  /* Unix variants all use the same ABI for x86-64  */
+#else
+  /* ---- AMD x86-64 ---- */
   SET_ENUM_VALUE(FFI_UNIX64);
 #endif
 
