--- a/node.gyp
+++ b/node.gyp
@@ -1144,6 +1144,7 @@
       'includes': [
         'node.gypi'
       ],
+      'libraries!':[ '-licui18n', '-licuuc', '-licudata', '-lcrypto', '-lssl', '-lz', '-lhttp_parser', '-luv', '-lnghttp2', '-lcares' ],
       'include_dirs': [
         'src',
         'tools/msvs/genfiles',
@@ -1353,6 +1354,7 @@
       'target_name': 'node_js2c',
       'type': 'executable',
       'toolsets': ['host'],
+      'libraries!':[ '-lcrypto', '-lssl', '-lz', '-lhttp_parser', '-luv', '-lnghttp2', '-lcares' ],
       'include_dirs': [
         'tools',
         'src',
