--- a/dist/bin/platforms/linux.js
+++ b/dist/bin/platforms/linux.js
@@ -26,17 +26,17 @@
     async install() {
         this.checkForRoot();
         await this.checkUser();
-        this.setupSudo();
+        // this.setupSudo();
         await this.hbService.portCheck();
         await this.hbService.storagePathCheck();
         await this.hbService.configCheck();
         try {
-            await this.createSystemdEnvFile();
-            await this.createSystemdService();
+            // await this.createSystemdEnvFile();
+            // await this.createSystemdService();
             await this.createRunPartsPath();
-            await this.reloadSystemd();
-            await this.enableService();
-            await this.start();
+            // await this.reloadSystemd();
+            // await this.enableService();
+            // await this.start();
             await this.hbService.printPostInstallInstructions();
         }
         catch (e) {
@@ -397,4 +397,4 @@
     }
 }
 exports.LinuxInstaller = LinuxInstaller;
