--- a/dist/bin/hb-service.js
+++ b/dist/bin/hb-service.js
@@ -17,7 +17,7 @@ const tar = require("tar");
 const tcpPortUsed = require("tcp-port-used");
 const darwin_1 = require("./platforms/darwin");
 const freebsd_1 = require("./platforms/freebsd");
-const linux_1 = require("./platforms/linux");
+const linux_1 = require("./platforms/openwrt");
 const win32_1 = require("./platforms/win32");
 class HomebridgeServiceHelper {
     get logPath() {
