--- a/binding.gyp
+++ b/binding.gyp
@@ -68,5 +68,33 @@
                 },
               ],
               [
+                'target_arch=="mips"', {
+                  'variables': {
+                    'openssl_config_path': '<(nodedir)/deps/openssl/config/mips'
+                  }
+                }
+              ],
+              [
+                'target_arch=="mips64"', {
+                  'variables': {
+                    'openssl_config_path': '<(nodedir)/deps/openssl/config/mips64'
+                  }
+                }
+              ],
+              [
+                'target_arch=="mipsel"', {
+                  'variables': {
+                    'openssl_config_path': '<(nodedir)/deps/openssl/config/mipsel'
+                  }
+                }
+              ],
+              [
+                'target_arch=="mips64el"', {
+                  'variables': {
+                    'openssl_config_path': '<(nodedir)/deps/openssl/config/mips64el'
+                  }
+                }
+              ],
+              [
                 'target_arch=="ppc64"', {
                   'variables': {
