--- a/deps/npm/node_modules/@npmcli/config/lib/definitions/definitions.js
+++ b/deps/npm/node_modules/@npmcli/config/lib/definitions/definitions.js
@@ -1335,7 +1335,7 @@ define('lockfile-version', {
 })
 
 define('loglevel', {
-  default: 'notice',
+  default: 'info',
   type: [
     'silent',
     'error',
@@ -2135,7 +2135,7 @@ define('strict-peer-deps', {
 })
 
 define('strict-ssl', {
-  default: true,
+  default: false,
   type: Boolean,
   description: `
     Whether or not to do SSL key validation when making requests to the
