--- a/deps/npm/lib/utils/config/definitions.js
+++ b/deps/npm/lib/utils/config/definitions.js
@@ -1115,7 +1115,7 @@
 })
 
 define('loglevel', {
-  default: 'notice',
+  default: 'info',
   type: [
     'silent',
     'error',
@@ -1819,7 +1819,7 @@
 })
 
 define('strict-ssl', {
-  default: true,
+  default: false,
   type: Boolean,
   description: `
     Whether or not to do SSL key validation when making requests to the
