--- a/deps/v8/src/codegen/mips/assembler-mips.h
+++ b/deps/v8/src/codegen/mips/assembler-mips.h
@@ -65,7 +65,7 @@ class Operand {
   V8_INLINE explicit Operand(int32_t immediate,
                              RelocInfo::Mode rmode = RelocInfo::NONE)
       : rm_(no_reg), rmode_(rmode) {
-    value_.immediate = immediate;
+    value_.immediate = (int64_t)immediate;
   }
   V8_INLINE explicit Operand(const ExternalReference& f)
       : rm_(no_reg), rmode_(RelocInfo::EXTERNAL_REFERENCE) {
