--- a/binding.gyp
+++ b/binding.gyp
@@ -21,7 +21,7 @@
         "target_conditions": [
           ["OS not in 'ios mac'", {
             # Avoid defining _FORTIFY_SOURCE on Darwin
-            "defines+": ["_FORTIFY_SOURCE=2"]
+            # "defines+": ["_FORTIFY_SOURCE=2"]
           }],
           ["OS not in 'win ios mac aix'", {
             # On Darwin with Xcode CLT/LLVM, "-fvisibility=hidden" hide all symbols that
