--- a/tools/icu/icu-generic.gyp
+++ b/tools/icu/icu-generic.gyp
@@ -531,6 +531,7 @@
       'target_name': 'genrb',
       'type': 'executable',
       'toolsets': [ 'host' ],
+      'libraries!':[ '-lcrypto', '-lssl', '-lz' ],
       'dependencies': [ 'icutools' ],
       'sources': [
         '<@(icu_src_genrb)'
@@ -547,6 +548,7 @@
       'target_name': 'iculslocs',
       'toolsets': [ 'host' ],
       'type': 'executable',
+      'libraries!':[ '-lcrypto', '-lssl', '-lz' ],
       'dependencies': [ 'icutools' ],
       'sources': [
         'iculslocs.cc',
@@ -559,6 +561,7 @@
       'target_name': 'icupkg',
       'toolsets': [ 'host' ],
       'type': 'executable',
+      'libraries!':[ '-lcrypto', '-lssl', '-lz' ],
       'dependencies': [ 'icutools' ],
       'sources': [
         '<@(icu_src_icupkg)',
@@ -570,6 +573,7 @@
       'target_name': 'genccode',
       'toolsets': [ 'host' ],
       'type': 'executable',
+      'libraries!':[ '-lcrypto', '-lssl', '-lz' ],
       'dependencies': [ 'icutools' ],
       'sources': [
         '<@(icu_src_genccode)',
