--- a/deps/v8/src/sampler.cc
+++ b/deps/v8/src/sampler.cc
@@ -405,9 +405,9 @@
   state.sp = reinterpret_cast<Address>(mcontext.gregs[29]);
   state.fp = reinterpret_cast<Address>(mcontext.gregs[30]);
 #elif V8_HOST_ARCH_PPC
-  state.pc = reinterpret_cast<Address>(ucontext->uc_mcontext.regs->nip);
-  state.sp = reinterpret_cast<Address>(ucontext->uc_mcontext.regs->gpr[PT_R1]);
-  state.fp = reinterpret_cast<Address>(ucontext->uc_mcontext.regs->gpr[PT_R31]);
+  state.pc = reinterpret_cast<Address>(ucontext->uc_mcontext.gregs[32]);
+  state.sp = reinterpret_cast<Address>(ucontext->uc_mcontext.gregs[1]);
+  state.fp = reinterpret_cast<Address>(ucontext->uc_mcontext.gregs[31]);
 #endif  // V8_HOST_ARCH_*
 #elif V8_OS_MACOSX
 #if V8_HOST_ARCH_X64
