--- a/tools/icu/icu-generic.gyp
+++ b/tools/icu/icu-generic.gyp
@@ -518,6 +518,7 @@
       'target_name': 'genrb',
       'type': 'executable',
       'toolsets': [ 'host' ],
+      'libraries!':[ '-lcrypto', '-lssl', '-lz' ],
       'dependencies': [ 'icutools' ],
       'sources': [
         '<@(icu_src_genrb)'
@@ -535,6 +536,7 @@
       'toolsets': [ 'host' ],
       'type': 'executable',
       'dependencies': [ 'icutools' ],
+      'libraries!':[ '-lcrypto', '-lssl', '-lz' ],
       'sources': [
         'iculslocs.cc',
         'no-op.cc',
@@ -546,6 +548,7 @@
       'target_name': 'icupkg',
       'toolsets': [ 'host' ],
       'type': 'executable',
+      'libraries!':[ '-lcrypto', '-lssl', '-lz' ],
       'dependencies': [ 'icutools' ],
       'sources': [
         '<@(icu_src_icupkg)',
@@ -557,6 +560,7 @@
       'target_name': 'genccode',
       'toolsets': [ 'host' ],
       'type': 'executable',
+      'libraries!':[ '-lcrypto', '-lssl', '-lz' ],
       'dependencies': [ 'icutools' ],
       'sources': [
         '<@(icu_src_genccode)',
