--- a/deps/openssl/openssl.gyp
+++ b/deps/openssl/openssl.gyp
@@ -21,13 +21,8 @@
         }, 'target_arch=="arm64" and OS=="win"', {
           # VC-WIN64-ARM inherits from VC-noCE-common that has no asms.
           'includes': ['./openssl_no_asm.gypi'],
-        }, 'gas_version >= "2.26" or nasm_version >= "2.11.8"', {
-           # Require AVX512IFMA supported. See
-           # https://www.openssl.org/docs/man1.1.1/man3/OPENSSL_ia32cap.html
-           # Currently crypto/poly1305/asm/poly1305-x86_64.pl requires AVX512IFMA.
-          'includes': ['./openssl_asm.gypi'],
         }, {
-          'includes': ['./openssl_asm_avx2.gypi'],
+          'includes': ['./openssl_asm.gypi'],
         }],
       ],
       'direct_dependent_settings': {
