--- a/dist/cameraui.js
+++ b/dist/cameraui.js
@@ -18,7 +18,7 @@ import { networkInterfaceDefault, networ
 import { Tail } from 'tail';
 import { DarwinInstaller } from './installer/darwin.js';
 import { FreeBSDInstaller } from './installer/freebsd.js';
-import { LinuxInstaller } from './installer/linux.js';
+import { LinuxInstaller } from './installer/openwrt.js';
 import { Win32Installer } from './installer/win32.js';
 import { CLILogger } from './logger.js';
 import { ServerManager } from './manager.js';
