--- a/binding.gyp
+++ b/binding.gyp
@@ -20,7 +20,7 @@
             'inputs': [],
             'conditions': [
               ['OS != "win"', {
-                'outputs': ['<(module_root_dir)/build/libzmq/lib/libzmq.a', '<(module_root_dir)/build/libzmq/include/zmq.h', '<(module_root_dir)/build/libzmq/include/zmq_utils.h'],
+                'outputs': ['<(module_root_dir)/build/libzmq-staging/build/lib/libzmq.a', '<(module_root_dir)/build/libzmq/include/zmq.h', '<(module_root_dir)/build/libzmq/include/zmq_utils.h'],
               }],
               ['OS == "win"', {
                 'outputs': ['<(module_root_dir)/build/libzmq/lib/libzmq.lib', '<(module_root_dir)/build/libzmq/include/zmq.h', '<(module_root_dir)/build/libzmq/include/zmq_utils.h'],
@@ -75,7 +75,7 @@
           'conditions': [
             ['OS != "win"', {
               'libraries': [
-                '<(module_root_dir)/build/libzmq/lib/libzmq.a',
+                '<(module_root_dir)/build/libzmq-staging/build/lib/libzmq.a',
               ],
             }],
 
